plugins/module_utils/omevv_utils/omevv_firmware_utils.py

```diff
@@ -254,3 +274,337 @@ def delete_firmware_repository_profile(self, profile_id):
+            dict: A dictionary representing the job schedule.
+        """
+        if days and time:
+            days_selected = set(days)
+
+            if "all" in days_selected:
+                return {
+                    "monday": True,
+                    "tuesday": True,
+                    "wednesday": True,
+                    "thursday": True,
+                    "friday": True,
+                    "saturday": True,
+                    "sunday": True,
+                    "time": time
+                }
+            else:
+                return {
+                    "monday": "monday" in days_selected,
+                    "tuesday": "tuesday" in days_selected,
+                    "wednesday": "wednesday" in days_selected,
+                    "thursday": "thursday" in days_selected,
+                    "friday": "friday" in days_selected,
+                    "saturday": "saturday" in days_selected,
+                    "sunday": "sunday" in days_selected,
+                    "time": time
+                }
+        return None
+
+    def get_current_job_schedule(self, profile_id, vcenter_uuid):
+        """
+        Retrieves the current job schedule for a baseline profile.
+
+        Args:
+            profile_id (str): The ID of the profile for which to retrieve the schedule.
+            vcenter_uuid (str): The UUID of the vCenter.
+
+        Returns:
+            dict: Job schedule details if available.
+        """
+        resp = self.omevv.invoke_request(
+            "GET", DRIFT_URI.format(vcenter_uuid=vcenter_uuid) + "/" + str(profile_id)
+        )
+        return resp
+
+    def search_baseline_profile_name(self, data, profile_name):
+        """
+        Searches for a profile with the given name in the provided data.
+
+        Args:
+            data (list): A list of dictionaries representing profiles.
+            profile_name (str): The name of the profile to search for.
+
+        Returns:
+            dict: The dictionary representing the profile if found, or an empty dictionary if not found.
+        """
+        for d in data:
+            if d.get('name') == profile_name:
+                return d
+        return {}
+
+    def get_add_remove_group_ids(self, existing_profile, vcenter_uuid, cluster_names):
+        """Determine groups to add or remove based on the cluster names"""
+        current_group_ids = {group['omevv_groupID'] for group in existing_profile.get('clusterGroups', [])}
+
+        new_group_ids = set(self.get_group_ids_for_clusters(
+            vcenter_uuid=vcenter_uuid,
+            cluster_names=cluster_names
+        ))
+
+        # Determine the groups that need to be added or removed
+        add_group_ids = list(new_group_ids - current_group_ids)
+        remove_group_ids = list(current_group_ids - new_group_ids)
+
+        return add_group_ids, remove_group_ids
+
+    def get_create_payload_details(self, name, firmware_repo_id, group_ids, job_schedule, description=None):
```

Comment:
move this to module, don't write module specific code in utils.

Reply:
Done